assets/js/search-bar.js

```diff
@@ -139,38 +150,47 @@ function hideAutocomplete () {
 }
 
 let lastScrollTop = window.scrollY
-const topSearch = document.querySelector('.top-search')
+const topSearch = document.querySelector(TOP_SEARCH_SELECTOR)
 const sidebarMenu = document.getElementById('sidebar-menu')
 const backgroundLayer = document.querySelector('.background-layer')
 const scrollThreshold = 70 // Set a threshold for scroll, adjust as needed
+const searchInput = qs(SEARCH_INPUT_SELECTOR)
+const sidebarButton = qs('.sidebar-button')
 
 window.addEventListener('scroll', function () {
   const currentScroll = window.scrollY
 
-  // Add 'fixed' class when not at the top
-  if (currentScroll > scrollThreshold * 2) {
-    topSearch.classList.add('sm-fixed')
-    sidebarMenu.classList.add('sm-fixed')
-    backgroundLayer.classList.add('sm-fixed')
-  }
+  if (isTouchDevice()) {
+    // Add 'fixed' class when not at the top
+    if (currentScroll > scrollThreshold * 2) {
+      topSearch.classList.add('sm-fixed')
+      sidebarMenu.classList.add('sm-fixed')
+      backgroundLayer.classList.add('sm-fixed')
+    }
 
-  if (currentScroll === 0) {
-    // Remove 'fixed' class when at the top
-    topSearch.classList.remove('sm-fixed')
-    sidebarMenu.classList.remove('sm-fixed')
-    backgroundLayer.classList.remove('sm-fixed')
-  }
+    if (currentScroll === 0) {
+      // Remove 'fixed' class when at the top
+      topSearch.classList.remove('sm-fixed')
+      sidebarMenu.classList.remove('sm-fixed')
+      backgroundLayer.classList.remove('sm-fixed')
+    }
 
-  if (currentScroll > lastScrollTop && currentScroll > scrollThreshold) {
-    // Scrolling down and past the threshold
-    topSearch.classList.add('sm-hidden')
-    sidebarMenu.classList.add('sm-hidden')
-    backgroundLayer.classList.add('sm-hidden')
-  } else {
-    // Scrolling up or at the top of the page
-    topSearch.classList.remove('sm-hidden')
-    sidebarMenu.classList.remove('sm-hidden')
-    backgroundLayer.classList.remove('sm-hidden')
+    if (currentScroll > lastScrollTop && currentScroll > scrollThreshold) {
+      // Scrolling down and past the threshold
+      topSearch.classList.add('sm-hidden')
+      sidebarMenu.classList.add('sm-hidden')
+      backgroundLayer.classList.add('sm-hidden')
+    } else {
+      // Scrolling up or at the top of the page
+      topSearch.classList.remove('sm-hidden')
+      sidebarMenu.classList.remove('sm-hidden')
+      backgroundLayer.classList.remove('sm-hidden')
+    }
+  } else if (currentScroll !== lastScrollTop) {
+    topSearch.classList.remove('sticky')
```

Comment:
I was going to extract these into a `closeSearchbar` function but it's only called once and it would require passing in the DOM elements as dependencies otherwise we'd running three DOM queries in a scroll callback which would be worse than this run-on sentence :)